datafusion/tests/sql.rs

```diff
@@ -5426,6 +5426,67 @@ async fn qualified_table_references() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn qualified_table_references_and_fields() -> Result<()> {
+    let mut ctx = ExecutionContext::new();
+
+    let c1: StringArray = vec!["foofoo", "foobar", "foobaz"]
+        .into_iter()
+        .map(Some)
+        .collect();
+    let c2: Int64Array = vec![1, 2, 3].into_iter().map(Some).collect();
+
+    let batch = RecordBatch::try_from_iter(vec![
+        ("f.c1", Arc::new(c1) as ArrayRef),
+        //  evil -- use the same name as the table
+        ("test.c2", Arc::new(c2) as ArrayRef),
+    ])?;
+
+    let table = MemTable::try_new(batch.schema(), vec![vec![batch]])?;
+    ctx.register_table("test", Arc::new(table))?;
+
+    // referring to the unquoted column is an error
+    let sql = format!(r#"SELECT f1.c1 from test"#);
+    let error = ctx.create_logical_plan(&sql).unwrap_err();
+    assert_contains!(
+        error.to_string(),
+        "No field named 'f1.c1'. Valid fields are 'test.f.c1', 'test.test.c2'"
+    );
+
+    // however, enclosing it in double quotes is ok
+    let sql = format!(r#"SELECT "f.c1" from test"#);
+    let actual = execute_to_batches(&mut ctx, &sql).await;
+    let expected = vec![
+        "+--------+",
+        "| f.c1   |",
```

Comment:
i wonder if it is okay to create column `....`

Reply:
I think it is "ok" from the point of view of DataFusion should run the query. I don't think anyone should actually do it 😆 

Added a test for this case in 3e4418c67